### PR TITLE
feat: `--output` for oras manifest `index update`

### PR DESCRIPTION
**Pull request by wangxiaoxuan273**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
part of #1053 

**Please check the following list**:
- [x]  Does the affected code have corresponding tests, e.g. unit test, E2E test?
- [ ]  Does this change require a documentation update?
- [ ]  Does this introduce breaking changes that would require an announcement or bumping the major version?
- [ ]  Do all new files have an appropriate license header?

### CODE REVIEW

**Comment by qweeah** [comment]
Please make sure `--output -` make other stdout output silent. See https://github.com/oras-project/oras/issues/1503. Doesn't necessary need to be done in this PR.

**Comment by wangxiaoxuan273** [comment]
> Please make sure `--output -` make other stdout output silent. See #1503. Doesn't necessarily need to be done in this PR.

I'd like to do it in a future PR.

File: cmd/oras/root/manifest/index/update.go [hunk 1]
```diff
@@ -64,9 +67,18 @@ Example - Merge manifests from the index 'v2-windows' to the index 'v2':
 
 Example - Update an index and tag the updated index as 'v2.1.0' and 'v2':
   oras manifest index update localhost:5000/hello@sha256:99e4703fbf30916f549cd6bfa9cdbab614b5392fbe64fdee971359a77073cdf9 --add linux-amd64 --tag "v2.1.0" --tag "v2"
+
+Example - Update an index and save it locally to index.json, auto push will be disabled:
```
Comment (Wwwsylvia) [inline cmd/oras/root/manifest/index/update.go:71]: nit: I think "auto push will be disabled" looks a bit too-much. How about something like 
> Update the index and save it locally to index.json without pushing it.

?

Comment (wangxiaoxuan273) [reply cmd/oras/root/manifest/index/update.go:71]: I think `auto push will be disabled` is clearer and emphasizes that there will be no push when `--output` is used.

Comment (shizhMSFT) [reply cmd/oras/root/manifest/index/update.go:71]: We can update it later since the current wording is consistent with `index create`.

**Review by shizhMSFT — APPROVED** [review]
LGTM

**Pull request merged**